sycl/doc/extensions/ExtendedAtomics/SYCL_INTEL_extended_atomics.asciidoc

```diff
@@ -113,389 +115,399 @@ In SYCL 1.2.1, support for floating-point types is limited to the +load+, +store
-  T fetch_sub(T operand, memory_order memoryOrder =
-    memory_order::relaxed);
-
-  /* Available only when: T != float */
-  T fetch_and(T operand, memory_order memoryOrder =
-    memory_order::relaxed);
-
-  /* Available only when: T != float */
-  T fetch_or(T operand, memory_order memoryOrder =
-    memory_order::relaxed);
-
-  /* Available only when: T != float */
-  T fetch_xor(T operand, memory_order memoryOrder =
-    memory_order::relaxed);
-
-  /* Available only when: T != float */
-  T fetch_min(T operand, memory_order memoryOrder =
-    memory_order::relaxed);
-
-  /* Available only when: T != float */
-  T fetch_max(T operand, memory_order memoryOrder =
-    memory_order::relaxed);
+inline constexpr memory_order memory_order_relaxed = memory_order::relaxed;
+inline constexpr memory_order memory_order_acquire = memory_order::acquire;
+inline constexpr memory_order memory_order_release = memory_order::release;
+inline constexpr memory_order memory_order_acq_rel = memory_order::acq_rel;
+inline constexpr memory_order memory_order_seq_cst = memory_order::seq_cst;
+
+// Exposition only
+template <memory_order ReadModifyWriteOrder>
+struct memory_order_traits;
+
+template <>
+struct memory_order_traits<memory_order::relaxed> {
+  static constexpr memory_order read_order = memory_order::relaxed;
+  static constexpr memory_order write_order = memory_order::relaxed;
 };
-} // namespace sycl
-} // namespace cl
-----
 
-==== To:
+template <>
+struct memory_order_traits<memory_order::acq_rel> {
+  static constexpr memory_order read_order = memory_order::acquire;
+  static constexpr memory_order write_order = memory_order::release;
+};
 
-[source,c++]
-----
-namespace cl {
-namespace sycl {
-enum class memory_order : int {
-  relaxed
+template <>
+struct memory_order_traits<memory_order::seq_cst> {
+  static constexpr memory_order read_order = memory_order::seq_cst;
+  static constexpr memory_order write_order = memory_order::seq_cst;
 };
-namespace intel {
-template <typename T, access::address_space addressSpace =
-  access::address_space::global_space>
-class atomic {
+
+template <typename T, memory_order DefaultOrder, access::address_space Space>
+class atomic_ref {
  public:
 
-  atomic(multi_ptr<T, addressSpace> ptr);
-  atomic(const atomic&);
-  atomic& operator=(const atomic&) = delete;
+  using value_type = T;
+  static constexpr size_t required_alignment = /* implementation-defined */;
+  static constexpr bool is_always_lock_free = /* implementation-defined */;
+  static constexpr memory_order default_read_order = memory_order_traits<DefaultOrder>::read_order;
+  static constexpr memory_order default_write_order = memory_order_traits<DefaultOrder>::write_order;
+  static constexpr memory_order default_read_modify_write_order = DefaultOrder;
 
   bool is_lock_free() const;
 
-  void store(T operand, memory_order order =
-    memory_order::relaxed);
+  explicit atomic_ref(T&);
+  atomic_ref(const atomic_ref&);
```

Comment:
This constructor is not documented in the corresponding table of member functions

Reply:
Thanks, I missed this.  Added.